shawmut/bin/auto_offd.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env
+
+import time
+from datetime import datetime
+from subprocess import call
+import os.path
+import urllib2
+import json
+
+SCHEDULER_INTERVAL = 5
+ADDRESSES = [ '10.0.1.16', '192.168.13.6' ]
+
+# TODO capture timestamp instead of away_flag
+# TODO only turn on lights on night time
+
+def log(msg):
+    print "%s: %s" % (datetime.now().strftime('%Y-%m-%d %H:%M:%S'), msg)
+
+def check_if_home():
+    for ip in ADDRESSES:
+        DEVNULL = open(os.devnull, 'w') # TODO move outside loop
+        ip_status = call(['ping', ip, '-q', '-w', '1', '-c', '1'], stdout=DEVNULL, close_fds=True)
+        log("Found status of %s pinging %s" %(ip_status, ip))
+        if ip_status == 0:
+            return True
+    return False
+
+def have_guests():
+    os.path.isfile('/etc/shawmut/have_guests')
+
+def light_data():
+    data = urllib2.urlopen('http://localhost:5000/api/environment').read()
+    return json.loads(data)
+
+def off_lights():
+    return list(name for (name, data) in light_data().iteritems() if data['state'] == 0)
+
+def on_lights():
+    return list(name for (name, data) in light_data().iteritems() if data['state'] == 1)
+
+def toggle_lights(lights):
+    for l in lights:
+        urllib2.urlopen("http://localhost:5000/api/device/%s" %l, '{"state":"toggle"}')
```

Comment:
Why toggle? I think this function should take a state parameter ("on" or "off") and set the state of the lights to that param. "toggle" will break if the lights are already on/off.